sdk/cosmos/azure-cosmos/azure/cosmos/aio/user.py

```diff
@@ -0,0 +1,295 @@
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# pylint: disable=missing-client-constructor-parameter-credential,missing-client-constructor-parameter-kwargs
+
+"""Create, read, update and delete users in the Azure Cosmos DB SQL API service.
+"""
+
+from typing import Any, List, Dict, Union, cast, Iterable, Optional
+
+from azure.core.tracing.decorator_async import distributed_trace_async
+from azure.core.tracing.decorator import distributed_trace
+
+from ._cosmos_client_connection_async import CosmosClientConnection
+from .._base import build_options
+from ..permission import Permission
+
+
+class UserProxy(object):
+    """An interface to interact with a specific user.
+
+    This class should not be instantiated directly. Instead, use the
+    :func:`DatabaseProxy.get_user_client` method.
+    """
+
+    def __init__(self, client_connection, id, database_link, properties=None):  # pylint: disable=redefined-builtin
+        # type: (CosmosClientConnection, str, str, Dict[str, Any]) -> None
+        self.client_connection = client_connection
+        self.id = id
+        self.user_link = u"{}/users/{}".format(database_link, id)
+        self._properties = properties
+
+    def __repr__(self):
+        # type () -> str
+        return "<UserProxy [{}]>".format(self.user_link)[:1024]
+
+    def _get_permission_link(self, permission_or_id):
+        # type: (Union[Permission, str, Dict[str, Any]]) -> str
+        if isinstance(permission_or_id, str):
+            return u"{}/permissions/{}".format(self.user_link, permission_or_id)
+        try:
+            return cast("Permission", permission_or_id).permission_link
+        except AttributeError:
+            pass
+        return u"{}/permissions/{}".format(self.user_link, cast("Dict[str, str]", permission_or_id)["id"])
+
+    async def _get_properties(self):
+        # type: () -> Dict[str, Any]
+        if self._properties is None:
+            self._properties = await self.read()
+        return self._properties
+
+    @distributed_trace_async
+    async def read(self, **kwargs):
+        # type: (Any) -> Dict[str, Any]
+        """Read user propertes.
+
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :returns: A dictionary of the retrieved user properties.
+        :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: If the given user couldn't be retrieved.
+        :rtype: dict[str, Any]
+        """
+        request_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
```

Comment:
we need to check what the use case for response_hook is.
If we are already returning the result why do we need resonse_hook